enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+a relatively advanced strategy to employ: only updating the control-plane and leaving worker-nodes as-is.
+It is possible by pausing machine config pools, but this is certainly not an intuitive step for users. Farther back 
+in OpenShift 4's history, the strategy was not even safe to perform since it could lead to worker-node 
+certificates to expiring. 
+
+By separating the control-plane and worker-node updates into two separate steps, we provide a clear
+and intuitive method of deferring worker-node updates: not initiating them. Leaving this to the user's
+discretion, within safe skew-bounds, gives them the flexibility to make the right choices for their
+unique circumstances.
+
+#### Enhancing Operational Control
+The preceding section delved deeply into a motivation for Change Management / Maintenance Schedules based on our desire to 
+separate control-plane and worker-node updates without increasing operational burden on end-users. However,
+Change Management, by providing control over exactly when updates & material changes to nodes in
+the cluster can be initiated, provide value irrespective of this strategic direction. The benefit of
+controlling exactly when changes are applied to critical systems is universally appreciated in enterprise 
+software. 
+
+Since these are such well established principles, I will summarize the motivation as helping
+OpenShift meet industry standard expectations with respect to limiting potentially disruptive change 
+outside well planned time windows. 
+
+It could be argued that rigorous and time sensitive management of OpenShift cluster API resources could prevent
+unplanned material changes, but Change Management / Maintenance Schedules introduce higher level, platform native, and more 
+intuitive guard rails. For example, consider the common pattern of a gitops configured OpenShift cluster.
+If a user wants to introduce a change to a MachineConfig, it is simple to merge a change to the
+resource without appreciating the fact that it will trigger a rolling reboot of nodes in the cluster.
+
+Trying to merge this change at a particular time of day and/or trying to pause and unpause a 
+MachineConfigPool to limit the impact of that merge to a particular time window requires
+significant forethought by the user. Even with that forethought, if an enterprise wants 
+changes to only be applied during weekends, additional custom mechanics would need
+to be employed to ensure the change merged during the weekend without needing someone present.
+
+Contrast this complexity with the user setting a Change Management / Maintenance Schedule on the cluster. The user
+is then free to merge configuration changes and gitops can apply those changes to OpenShift
+resources, but material change to the cluster will not be initiated until a time permitted
+by the Maintenance Schedule. Users do not require special insight into the implications of
```

Comment:
Good catch. I need to make this explicit -- there is no guarantee. The schedule assures that no material change is initiated, but the state of rebooted nodes / new nodes is dependent on the profile and state of rollout. For example, in standalone, if a new release payload is targeted by the CVO, but cluster change management is paused, the CVO will not roll out machine configs which ultimately inform new nodes joining the cluster. Thus, new nodes joining the cluster will receive the old configuration.
If, however, the MCP is halfway through updating old nodes when change management is paused, new nodes joining the cluster will receive the new configuration.

Reply:
Can we make that more deterministic somehow? How hard would it be to apply a simple rule like "new nodes will always be deployed with the latest config" or "new nodes will always be deployed with a consistent configuration"?

Reply:
> "new nodes will always be deployed with a consistent configuration"

That isn't hard. There will be a distinct moment where the MCS will start serving the new vs the old. 

> "new nodes will always be deployed with the latest config"

Since we need to implement this for multiple controllers, the safe bet is to leave that choice up to the implementation. Intuitively, any control loop can be paused, so, as written, the enhancement should be applicable across any controller. However, to provide this guarantee, we would need to say something like "no external sign of inconsistency can be visible until reconciliation is complete".

There is inherent friction with eventual consistency in that statement.

Consider the questions it raises in standalone:
- In a paused state with MachineConfig changes pending, there is a clear difference between "new" vs "old" config. Since the "new" is not incorporated into an MCP yet, new machines booted into the cluster must necessarily receive the old.
- Once the CVO unpauses and introduces a MachineConfig update, the MCP will be updated. If we pause at this point, what should a new machine receive? If you say "new", we've violated the proposed contract.
- Once the MCO has updated 75% of the machines, should it still be serving the old config? Is consistency more important than the inevitable prolonging of the update when those new nodes running old config need to be drained & rebooted? Do we start serving new config while they newish nodes are being rebooted (if we do, its possible that we will serve old config indefinitely if new machines constantly scale into and out of the cluster)?

There are detailed answers we _could_ give here that would make the standalone update process subtly more predictable. But is CAPI going to implement this contract? Would <future upstream controller>?

In short, I think all can reasonably insist upon is the ability to pause your progress towards consistency. If we try to define what the behavior of the system must be when we pause you, things get complex.

Reply:
I can't find any documented contract for the behavior in the existing pausing primitives provided by MCP or CAPI, so I'll need to leave this for SMEs to weigh in on the what is possible here.